### PR TITLE
feat: add character limit constraint to player name inputs

### LINKED ISSUE #267: [Feature] Add Character Limit Constraint to Player Name Inputs
### Description
Currently, the `PlayerNamesModal` component allows players to enter names of any length. This can cause layout overflow issues or UI misalignment when excessively long names are entered.

### Proposed Solution
Add a character limit constraint (e.g., **max 15 characters**) for both Player 1 and Player 2 input fields in the `PlayerNamesModal` component.

### Acceptance Criteria
- [x] Player name inputs should not accept more than 15 characters.  
- [ ] A small toast or inline error should appear if the user tries to exceed the limit.  (optional)
- [x] The limit should be configurable. 
- [x] Ensure UI layout remains responsive with max-length names.

### Relevant File
`/components/modals/PlayerNamesModal.tsx`

### PR DESCRIPTION
**Pull request by aditya241104**
# Add Character Limit Constraint to Player Name Inputs

- Add MAX_PLAYER_NAME_LENGTH constant (15 characters)
- Implement validation in onChange handlers
- Add character counter display for both inputs
- Prevent typing beyond character limit
- Ensure UI remains responsive with max-length names

Closes #267

### Purpose of this PR:
This PR adds character limit constraints to player name inputs in the PlayerNamesModal component to prevent layout overflow issues and UI misalignment when excessively long names are entered. The implementation adds a configurable 15-character limit with visual feedback through character counters.

### How to test this PR manually:
1. Start the development server with `pnpm dev`
2. Navigate to the player names modal in the game
3. Try typing in the player name input fields
4. Verify that you cannot type more than 15 characters
5. Check that the character counter displays correctly (e.g., "12/15 characters")
6. Test with different name lengths (1 char, 15 chars, attempting 16+ chars)
7. Ensure the UI layout remains responsive and doesn't break
8. Verify both Player 1 and Player 2 inputs work identically

### Related Issue
Closes #267

## Category (replace the empty space inside brackets with x for tick)
- [ ] Bug
- [ ] Refactor
- [x] Enhancement/New Feature
- [ ] Tests
- [ ] Documentation

## Checklist 
- [x] I have read and reviewed each line of my Git/File Differences for this PR very carefully and I clearly understand the reason behind the code changes and decisions I have made
- [x] This PR is short and cant be decomposed further without breaking consistency (check ../docs/SHORT_PR.md for more information)
- [x] I have manually tested all the changes and indirect impacts these changes could have made
- [ ] I have added/updated automated testing scripts (check ../docs/TESTS_FOR_PR.md for more information)
- [x] I will keep my branch updated with main till this branch doesn't get merged
- [x] Documentations updated if applicable
- [x] Code reviewed for accessibility
- [x] I will check the review made by coderabbit (both actionable and nitpick) and will respond with atleast a short comment of why I disagree with it
- [x] I understand maintainers might not be able to help with general doubts like git, tech stack etc
- [x] I will post the doubts regarding repository/project in Discussions tab only, so that fellow contributors could help me
- [x] I understand that doubts related to an issue or PR will be made as comment to the same.
- [x] I do not have any existing open PR pending to be merged
- [ ] If I am making this PR as a part of any competition then I will name it and mention e-mail id registered with that competition

## Screenshots (if applicable)
image

## Summary by CodeRabbit

* **New Features**
  * Enforces a 15-character limit for player names and adds per-field character counters showing current vs. max.

* **Bug Fixes**
  * Keeps duplicate-name validation and toast cooldown behavior consistent; duplicate toasts are dismissed and cooldowns reset before submit.

* **UI**
  * Wraps each player input in a container and refines modal layout for clearer input feedback.

### CODE REVIEW

**Comment by AshmitSherigar** [comment]
@aditya241104 could you read the code rabbit nitpick and actionable comments (not compulsory to implement if you feel like it is needed you can commit them) 
and also the build is failing since you are not following the biome standards

**Comment by aditya241104** [comment]
@AshmitSherigar  Thank you for the feedback! I'm still new to open source contributions, so I appreciate your patience as I learn the proper standards. 

I'll fix the biome formatting issues and address the CodeRabbit suggestions right away. Thanks for guiding me through this process!

File: src/modals/PlayerNamesModal.tsx [hunk 1]
```diff
@@ -52,17 +70,31 @@ const PlayerNamesModal = ({
...
+							value={player2}
+							onChange={handlePlayer2Change}
+							placeholder="Player 2 Name"
+							maxLength={MAX_PLAYER_NAME_LENGTH}
+						/>
+						{/* ✅ Character counter */}
+						<div className="text-xs text-gray-600 mt-1">
+							{player2.length}/{MAX_PLAYER_NAME_LENGTH} characters
+						</div>
+					</div>
```
[the remark this answers is not in the document]
Comment (AshmitSherigar) [reply src/modals/PlayerNamesModal.tsx:81]: This would be another issue

**Comment by AshmitSherigar** [comment]
> @AshmitSherigar Thank you for the feedback! I'm still new to open source contributions, so I appreciate your patience as I learn the proper standards.
> 
> I'll fix the biome formatting issues and address the CodeRabbit suggestions right away. Thanks for guiding me through this process!

yea we are here to learn any doubt please do ask

**Comment by AshmitSherigar** [comment]
@aditya241104 
1) The color and font needs to be changed 
2) I think it would be better if the character limit text was placed right side corner under the inputs

**Comment by AshmitSherigar** [comment]
rest all looks fine and good and can be merged once the changes are done

**Comment by aditya241104** [comment]
@AshmitSherigar Sorry for the late reply! I was busy with some other work and couldn't get back to this sooner.

I've updated the character counter styling as requested:
- ✅ Changed the color to a lighter grey (`text-gray-500`)
- ✅ Moved the character counters to the right side using `text-right` alignment
- ✅ Fixed all Biome formatting issues to match project standards

I have to admit, I don't have much experience with UI/UX color schemas or design principles yet, so I went with a lighter grey as you suggested. If you'd prefer a different color or have any specific design system preferences for this project, please let me know and I'll be happy to adjust it!

Thanks for your patience as I learn the project standards and improve my design sense. 🙏

**Comment by AshmitSherigar** [comment]
> 

posistion looks fine .. great job but the color and font size needs to be more visible especially the font size

File: src/modals/PlayerNamesModal.tsx [hunk 2]
```diff
@@ -52,17 +70,31 @@ const PlayerNamesModal = ({
...
-						value={player1}
-						onChange={(e) => setPlayer1(e.target.value)}
-						placeholder="Player 1 Name"
-					/>
+					<div>
+						<PlayerInput
+							value={player1}
+							onChange={handlePlayer1Change}
+							placeholder="Player 1 Name"
+							maxLength={MAX_PLAYER_NAME_LENGTH}
```
Comment (Rakshitg600) [inline src/modals/PlayerNamesModal.tsx:62]: we are using this prop in PlayerInput component?

File: src/modals/PlayerNamesModal.tsx [hunk 3]
```diff
@@ -27,14 +30,29 @@ const PlayerNamesModal = ({
...
+	};
+
 	const handleSubmit = () => {
 		if (!canShowToast()) return;
 
 		if (player1.trim().toLowerCase() === player2.trim().toLowerCase()) {
 			toast("Player 1 and Player 2 cannot have the same name.", {
 				toastId: TOAST_IDS.PlayerNames.Duplicate,
 				autoClose: TOAST_DURATION,
-				onClose: resetCooldown, // reset cooldown if closed early
```
Comment (Rakshitg600) [inline src/modals/PlayerNamesModal.tsx:37]: how is removing this comment relevant to this PR ?

File: src/modals/PlayerNamesModal.tsx (hunk 2 above)
Comment (aditya241104) [reply src/modals/PlayerNamesModal.tsx:62]: Looking at the PlayerInput component, it does accept these props since it spreads `{...props}` to the underlying input element, so both `maxLength` and custom `onChange` handlers should work.

Comment (aditya241104) [reply src/modals/PlayerNamesModal.tsx:62]: However, I realize I over-complicated this by creating separate handler functions when the original inline approach was simpler and clearer.

**Comment by AshmitSherigar** [comment]
@aditya241104 could you also add test cases either add the test cases (we use vitest and react testing library) or jus add the excel sheet which contains the necessary test cases you feel would be required and i could bump up the level of the PR for gsssoc

**Comment by aditya241104** [comment]
@Rakshitg600 I sincerely apologize for the earlier confusion. I've now properly addressed your feedback

**Comment by aditya241104** [comment]
> @AshmitSherigar could you also add test cases either add the test cases (we use vitest and react testing library) or jus add the excel sheet which contains the necessary test cases you feel would be required and i could bump up the level of the PR for gsssoc

@Rakshitg600 Absolutely! I'd be happy to add comprehensive test cases using Vitest and React Testing Library. This is a great learning opportunity for me.

**Comment by AshmitSherigar** [comment]
> > @AshmitSherigar could you also add test cases either add the test cases (we use vitest and react testing library) or jus add the excel sheet which contains the necessary test cases you feel would be required and i could bump up the level of the PR for gsssoc
> 
> @Rakshitg600 Absolutely! I'd be happy to add comprehensive test cases using Vitest and React Testing Library. This is a great learning opportunity for me.

This is for reference

https://docs.google.com/spreadsheets/u/0/d/1vOvHeaZgbSwvP6H8QpC4nK9NhUS8zVnGX2oJuZLmIS4/htmlview

Make a spread sheet like this and add and you have freedom to make different issues for different test case

**Comment by aditya241104** [comment]
@AshmitSherigar @Rakshitg600 

## Test Plan Completed ✅

I've created a comprehensive test plan spreadsheet for the character limit feature as requested. The test plan includes **20 detailed test cases** covering all scenarios:

🔗 **Test Plan Spreadsheet**: https://docs.google.com/spreadsheets/d/1jytSvImsSEE_RhscQh1N57qtIg0ljmrGw5vhr3LRn8k/edit?usp=sharing

### Test Coverage Summary:
- **9 High Priority** tests (core functionality)
- **8 Medium Priority** tests (edge cases)  
- **3 Low Priority** tests (additional validations)

### Key Test Categories:
- ✅ Character limit validation (15-char max)
- ✅ Character counter display accuracy
- ✅ Both Player 1 & Player 2 input validation
- ✅ Paste behavior with long text
- ✅ Form submission integration
- ✅ UI layout responsiveness
- ✅ Special characters & edge cases

The test plan follows the same format as the reference spreadsheet and covers all scenarios that could affect the character limit functionality. Each test case includes clear steps and expected results.

**Next Steps**: I'm happy to create a separate issue for implementing these test cases using Vitest and React Testing Library as a future contribution, or someone else can take on the implementation if preferred.

Thanks for the guidance on proper testing practices! This has been a great learning experience. 🚀

**Comment by AshmitSherigar** [comment]
> @AshmitSherigar @Rakshitg600
> 
> ## Test Plan Completed ✅
> I've created a comprehensive test plan spreadsheet for the character limit feature as requested. The test plan includes **20 detailed test cases** covering all scenarios:
> 
> 🔗 **Test Plan Spreadsheet**: https://docs.google.com/spreadsheets/d/1jytSvImsSEE_RhscQh1N57qtIg0ljmrGw5vhr3LRn8k/edit?usp=sharing
> 
> ### Test Coverage Summary:
> * **9 High Priority** tests (core functionality)
> * **8 Medium Priority** tests (edge cases)
> * **3 Low Priority** tests (additional validations)
> 
> ### Key Test Categories:
> * ✅ Character limit validation (15-char max)
> * ✅ Character counter display accuracy
> * ✅ Both Player 1 & Player 2 input validation
> * ✅ Paste behavior with long text
> * ✅ Form submission integration
> * ✅ UI layout responsiveness
> * ✅ Special characters & edge cases
> 
> The test plan follows the same format as the reference spreadsheet and covers all scenarios that could affect the character limit functionality. Each test case includes clear steps and expected results.
> 
> **Next Steps**: I'm happy to create a separate issue for implementing these test cases using Vitest and React Testing Library as a future contribution, or someone else can take on the implementation if preferred.
> 
> Thanks for the guidance on proper testing practices! This has been a great learning experience. 🚀

Could you give me access so i can look into the test case and then you can start on it

**Comment by aditya241104** [comment]
@AshmitSherigar Done

**Comment by AshmitSherigar** [comment]
@aditya241104 

ok looks good all the test cases .. some test cases could be combined also but that is up to you! .. please make sure to add the `.xlsx`  in  `csv format` file also in the `__test__` folder where all the test code is there in this PR it self .. and the test code focus on higher priority as one PR and so on

**Comment by AshmitSherigar** [comment]
@aditya241104 this looks fine from the overview … will merge in a few minutes you can start the test cases open a new issues stating the test cases u r solving and once u put a pr update the csv file accordingly that is from no to yes

**Review by Rakshitg600 — APPROVED** [review]

**Comment by Rakshitg600** [comment]
looks cool to me @aditya241104 , I can see you have not ticked the competition point in the PR description so I am assuming you are not making this PR as a part of any open source competition. Thanks for your contribution, looking forward to more contributions from you, all the best !

**Comment by aditya241104** [comment]
Thanks @AshmitSherigar! I'll definitely open a new issue for the test cases and start working on implementing them. I'm writing tests for the first time, so it might take a bit longer as I'm currently learning how to write proper test cases with Vitest and React Testing Library, but I'm committed to getting it done properly. I'll make sure to update the CSV file from "no" to "yes" as I complete each test case. Really appreciate your patience as I learn these new skills!

**Comment by aditya241104** [comment]
@Rakshitg600 Thank you for the approval and kind words! Actually, I need to clarify something - I did open this PR before I knew about Hacktoberfest, but I did register for Hacktoberfest later. Since this PR was not originally created for the competition, I didn't check that box initially.

I believe this PR will automatically count toward Hacktoberfest since it's merged in October and the repo has the "hacktoberfest" label.

**Comment by AshmitSherigar** [comment]
> Thanks @AshmitSherigar! I'll definitely open a new issue for the test cases and start working on implementing them. I'm writing tests for the first time, so it might take a bit longer as I'm currently learning how to write proper test cases with Vitest and React Testing Library, but I'm committed to getting it done properly. I'll make sure to update the CSV file from "no" to "yes" as I complete each test case. Really appreciate your patience as I learn these new skills!

Yes good i will merge ur PR right now!

**Comment by AshmitSherigar** [comment]
> @Rakshitg600 Thank you for the approval and kind words! Actually, I need to clarify something - I did open this PR before I knew about Hacktoberfest, but I did register for Hacktoberfest later. Since this PR was not originally created for the competition, I didn't check that box initially.
> 
> I believe this PR will automatically count toward Hacktoberfest since it's merged in October and the repo has the "hacktoberfest" label.

Yes it will be counted for hacktoberfest

**Comment by AshmitSherigar** [comment]
@aditya241104 congrats on your first PR😊 !
